packages/@ourworldindata/grapher/src/controls/ActionButtons.tsx

```diff
@@ -0,0 +1,371 @@
+        const labelWidth = Bounds.forText(label, { fontSize: 13 }).width
+        return (
+            2 * PADDING_X +
+            12 + // icon width
+            PADDING_BETWEEN_ICON_AND_LABEL +
+            labelWidth
+        )
+    }
+
+    @computed private get downloadButtonWithLabelWidth(): number {
+        return ActionButtons.computeButtonWidth("Download")
+    }
+
+    @computed private get shareButtonWithLabelWidth(): number {
+        return ActionButtons.computeButtonWidth("Share")
+    }
+
+    @computed private get fullScreenButtonLabel(): string {
+        const { isInFullScreenMode } = this.manager
+        return isInFullScreenMode ? "Exit full-screen" : "Enter full-screen"
+    }
+
+    @computed private get fullScreenButtonWithLabelWidth(): number {
+        return ActionButtons.computeButtonWidth(this.fullScreenButtonLabel)
+    }
+
+    @computed private get exploreTheDataButtonWithLabelWidth(): number {
+        return ActionButtons.computeButtonWidth("Explore the data")
+    }
+
+    @computed private get downloadButtonWidth(): number {
+        const {
+            hasDownloadButton,
+            showButtonLabels,
+            downloadButtonWithLabelWidth,
+        } = this
+        if (!hasDownloadButton) return 0
+        if (!showButtonLabels) return BUTTON_WIDTH_ICON_ONLY
+        return downloadButtonWithLabelWidth
+    }
+
+    @computed private get shareButtonWidth(): number {
+        const { hasShareButton, showButtonLabels, shareButtonWithLabelWidth } =
+            this
+        if (!hasShareButton) return 0
+        if (!showButtonLabels) return BUTTON_WIDTH_ICON_ONLY
+        return shareButtonWithLabelWidth
+    }
+
+    @computed private get fullScreenButtonWidth(): number {
+        const {
+            hasFullScreenButton,
+            showButtonLabels,
+            fullScreenButtonWithLabelWidth,
+        } = this
+        if (!hasFullScreenButton) return 0
+        if (!showButtonLabels) return BUTTON_WIDTH_ICON_ONLY
+        return fullScreenButtonWithLabelWidth
+    }
+
+    // the "Explore the data" button is never shown without a label
+    @computed private get exploreTheDataButtonWidth(): number {
+        const { hasExploreTheDataButton, exploreTheDataButtonWithLabelWidth } =
+            this
+        if (!hasExploreTheDataButton) return 0
+        return exploreTheDataButtonWithLabelWidth
+    }
+
+    @action.bound toggleShareMenu(): void {
+        this.manager.isShareMenuActive = !this.manager.isShareMenuActive
+    }
+
+    @action.bound toggleFullScreenMode(): void {
+        this.manager.isInFullScreenMode = !this.manager.isInFullScreenMode
+    }
+
+    @computed private get hasDownloadButton(): boolean {
+        return true
+    }
+
+    @computed private get hasShareButton(): boolean {
+        return !this.manager.hideShareTabButton
+    }
+
+    @computed private get hasFullScreenButton(): boolean {
+        return !this.manager.isInIFrame
```

Comment:
I don't think it makes sense to offer full-screen mode for charts that are embedded in an iframe. But that means that charts in gdocs won't show the full-screen button when previewed.. (since the gdoc preview is embedded via an iframe)